src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/SqlRootExpression.cs

```diff
@@ -13,43 +13,41 @@
 namespace Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions
 {
     /// <summary>
-    /// The root of a search expression tree that will be translated to a SQL command.
-    /// It is organized as a set of "normalized" table expression predicates and a set
-    /// of "denormalized" predicates. The normalized predicates are over a search parameter
-    /// table, whereas denormalized predicates are applied to the Resource table directly. Some
-    /// of them can be applied to search parameter tables as well.
+    /// The root of a search expression tree that will be translated to a SQL batch.
+    /// It is organized as a set of expressions that are over search parameter tables (<see cref="TableExpressions"/>) and a set
+    /// of expressions over the columns on the Resource table (<see cref="ResourceExpressions"/>).
     /// </summary>
     internal class SqlRootExpression : Expression
     {
-        public SqlRootExpression(IReadOnlyList<TableExpression> tableExpressions, IReadOnlyList<Expression> denormalizedExpressions)
+        public SqlRootExpression(IReadOnlyList<TableExpression> tableExpressions, IReadOnlyList<SearchParameterExpressionBase> resourceExpressions)
         {
             EnsureArg.IsNotNull(tableExpressions, nameof(tableExpressions));
-            EnsureArg.IsNotNull(denormalizedExpressions, nameof(denormalizedExpressions));
+            EnsureArg.IsNotNull(resourceExpressions, nameof(resourceExpressions));
 
             TableExpressions = tableExpressions;
-            DenormalizedExpressions = denormalizedExpressions;
+            ResourceExpressions = resourceExpressions;
         }
 
         public IReadOnlyList<TableExpression> TableExpressions { get; }
```

Comment:
>Table [](start = 46, length = 5)

nit: maybe it's worth to rename this one as well. Table is such a generic name, plus Resource table is also technically table.
`SearchParamExpressions` maybe?

I would also add summary 
```
<summary>
List of expression performed on various search parameter tables (e.g. TokenSearchParam, NumberSearchParam, etc)
</summary>
``` for this property and
```
<summary>
List of expression applied to directly to Resource table.
<summary>
``` for ResourceExpressions

Reply:
Ok and keep the `TableExpression` type as-is?

Reply:
Naming is tricky. -_-"
I wouldn't mind to rename it to `SearchParamTableExpression` as type and `SearchParamTableExpressions`, but there is comment on top of `TableExpression`: `An expression over a search param or **compartment** table`.

It's ok-ish the way it is right now, but I would still add summary comments for public properties.


---
In reply to: [553618664](https://github.com/microsoft/fhir-server/pull/1569#discussion_r553618664) [](ancestors = 553618664)

Reply:
Yeah. TableExpressions match 1:1 with the CTEs that we generate, so that's where the type name comes from. But I think `SearchParamTableExpressions` and `ResourceTableExpressions` could work. (I think we can address the compartment exception with a comment). What do you think?

Reply:
Sounds good to me.

---
In reply to: [553631371](https://github.com/microsoft/fhir-server/pull/1569#discussion_r553631371) [](ancestors = 553631371)

Reply:
Ok, I've done this in https://github.com/microsoft/fhir-server/pull/1569/commits/5eebddf2a917f25193a5a0f755e003ccf78e7855